### PR TITLE
Added Truncate and Wrap as new features on the Transform dialogue.

### PR DESCRIPTION
**Pull request by EstherNjeriLiberatta**
Fixes issue #6890
@africanmathsinitiative/developers @rdstern this is ready for review. Thanks.

### CODE REVIEW

**Review by rdstern — APPROVED** [review]
@EstherNjeriLiberatta this is excellent.  The only very minor point is that the Pad option (and also words (that you didn't look at) don't "remember" the settings of the string.  They always return to Space ( ).
Happy to approve at the same time and hope @shadrackkibet likes what you have done.

**Comment by EstherNjeriLiberatta** [comment]
@rdstern I looked into the problem. Requesting @lloyddewit to look into it. It is caused by `autoTranslate`. Thanks

image

File: instat/dlgTransformText.vb [hunk 1]
```diff
@@ -62,40 +62,55 @@ Public Class dlgTransformText
...
         ucrInputTo.AddFunctionNamesCondition("Upper", "str_to_upper")
         ucrInputTo.AddFunctionNamesCondition("Title", "str_to_title")
+        ucrInputTo.AddFunctionNamesCondition("Sentence", "str_to_sentence")
         ucrInputTo.SetLinkedDisplayControl(lblTo)
         ucrInputTo.SetDropDownStyleAsNonEditable()
 
         'rdoPad
         ucrPnlOperation.AddToLinkedControls(ucrInputPad, {rdoPad}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
-        ucrPnlOperation.AddToLinkedControls(ucrNudWidth, {rdoPad}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=1)
+        ucrPnlOperation.AddToLinkedControls(ucrNudWidth, {rdoPad}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, objNewDefaultState:=20, bNewLinkedChangeToDefaultState:=True)
```
Comment (Ivanluv) [inline instat/dlgTransformText.vb:97]: you don't need `objNewDefaultState:=20, bNewLinkedChangeToDefaultState:=True` when adding the parameters to the functions within the `setRDefaults()`.Also get rid of `bNewLinkedAddRemoveParameter:=True ` to  maintain previous state

File: instat/dlgTransformText.vb [hunk 2]
```diff
@@ -62,40 +62,55 @@ Public Class dlgTransformText
...
+        ucrPnlOperation.AddToLinkedControls(ucrNudWidthTrunc, {rdoTruncate}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, objNewDefaultState:=20, bNewLinkedChangeToDefaultState:=True)
```
Comment (Ivanluv) [inline instat/dlgTransformText.vb:98]: Do the same here

File: instat/dlgTransformText.vb [hunk 3]
```diff
@@ -62,40 +62,55 @@ Public Class dlgTransformText
...
+        ucrPnlOperation.AddToLinkedControls(ucrNudWidthWrap, {rdoWrap}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, objNewDefaultState:=40, bNewLinkedChangeToDefaultState:=True)
```
Comment (Ivanluv) [inline instat/dlgTransformText.vb:99]: Do the same here

Comment (EstherNjeriLiberatta) [reply instat/dlgTransformText.vb:99]: Okay

**Comment by EstherNjeriLiberatta** [comment]
@Ivanluv I've made the requested changes. This ready for review.

**Review by Ivanluv — DISMISSED** [review]

**Comment by EstherNjeriLiberatta** [comment]
@rdstern @africanmathsinitiative/developers This is ready for review.

**Review by rdstern — APPROVED** [review]

**Comment by shadrackkibet** [comment]
@EstherNjeriLiberatta could you improve the PR title? **Improved the Prepare >Column:Text > Transform Dialogue** is too general. Add some specific details to specify what changed.

**Comment by EstherNjeriLiberatta** [comment]
@shadrackkibet Done. I hope it's better? 
I also made a few changes on the Case and Pad options.

File: instat/dlgTransformText.vb [hunk 4]
```diff
@@ -288,8 +329,12 @@ Public Class dlgTransformText
                 Else
                     ucrBase.OKEnabled(False)
                 End If
+            ElseIf rdoTruncate.Checked Then
+                ucrBase.OKEnabled(True)
             ElseIf rdoTrim.Checked Then
                 ucrBase.OKEnabled(True)
+            ElseIf rdoWrap.Checked Then
+                ucrBase.OKEnabled(True)
```
Comment (shadrackkibet) [inline instat/dlgTransformText.vb:337]: Could make this one line to reduce the lines of code i.e `rdoTruncate.Checked OrElse rdoTrim.Checked OrElse rdoWrap.Checked Then`

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Comment by EstherNjeriLiberatta** [comment]
@shadrackkibet Done.

**Review by rdstern — APPROVED** [review]
Looks fine still

**Review by shadrackkibet — APPROVED** [review]

**Pull request merged**